ecommerce-ui/src/components/rentalForm/RentalForm.jsx

```diff
@@ -0,0 +1,172 @@
+  // Optional user inputs
+  description: false,
+  isSuperhost: false,
+};
+
+export default function RentalForm({ displayForm, userAddedRental }) {
+  const [formValues, setFormValues] = useState(defaultFormValues);
+  const [errorFlags, setErrorFlags] = useState(defaultErrorState);
+
+  const inputChangeHandler = (event) => {
+    const { name, value } = event.target;
+    setFormValues((currentState) => {
+      const updateState = { ...currentState };
+      if (name === "cost") {
+        updateState[name] = +value;
+      } else updateState[name] = value;
+      return updateState;
+    });
+  };
+
+  const submitHandler = (event) => {
+    event.preventDefault();
+
+    // For regex input tests
+    const emptyInput = /^\s*$/;
+
+    // Init values
+    const errorCheck = {
+      title: false,
+      houseType: false,
+      image: false,
+      imageAlt: false,
+      city: false,
+      country: false,
+      cost: false,
+      name: false,
+    };
+
+    // Complete error checks on form inputs
+    errorCheck.title = emptyInput.test(formValues.title) ? true : false;
+    errorCheck.houseType = emptyInput.test(formValues.houseType) ? true : false;
+    errorCheck.image = emptyInput.test(formValues.image) ? true : false;
+    errorCheck.imageAlt = emptyInput.test(formValues.imageAlt) ? true : false;
+    errorCheck.city = emptyInput.test(formValues.city) ? true : false;
+    errorCheck.country = emptyInput.test(formValues.country) ? true : false;
+    errorCheck.cost = emptyInput.test(formValues.cost) ? true : false;
+    errorCheck.name = emptyInput.test(formValues.name) ? true : false;
+
+    if (Object.values(errorCheck).includes(true)) {
+      setErrorFlags({
+        ...errorFlags,
+        title: [errorCheck.title, errorFlags.title[1]],
+        houseType: [errorCheck.houseType, errorFlags.houseType[1]],
+        image: [errorCheck.image, errorFlags.image[1]],
+        imageAlt: [errorCheck.imageAlt, errorFlags.imageAlt[1]],
+        city: [errorCheck.city, errorFlags.city[1]],
+        country: [errorCheck.country, errorFlags.country[1]],
+        cost: [errorCheck.cost, errorFlags.cost[1]],
+        name: [errorCheck.name, errorCheck.name[1]],
+      });
+      // Form will not be submitted
+      return;
+    }
+    // Checks pass, submit form
+    userAddedRental(formValues);
+    setFormValues(defaultFormValues);
+  };
+
+  return (
+    <DisplayCard cardColor="customBg.alt">
+      <form onSubmit={submitHandler}>
+        <Typography
+          sx={{ mt: 2, mb: 2, fontWeight: "bold" }}
+          variant="h6"
+          component="div"
+        >
+          Have a place to rent out? Enter your rental information below.
+        </Typography>
+        <Grid container spacing={2} direction="column" columns={12}>
+          <FormInputLine01
+            errorFlagsTitle={errorFlags.title}
+            value={formValues.title}
+            onChange={inputChangeHandler}
+          />
+
+          <FormInputLine02
```

Comment:
With these kinds of things, it's a best practice to instead of saying Line2, say what it actually does

<HouseTypeInput /> 

or something. It's easier to read

Reply:
Got it, thanks!